### ISSUE TITLE
Get rid of the `<<-` notation

### ISSUE THREAD

**Issue by gaborcsardi** [att/rcloud.rcap#133]
TL;DR: Let's update Rcloud to eliminate the `<<-` operators from the notebooks.

For two main reasons. One is that they are annoying: they do not allow easy code transfer. You cannot easily create a dashboard from an existing notebook, if you need to go over the whole thing and add double arrows to it. You also cannot just move code from your package or other notebook, into a dashboard. Or the other way.

The other reason is actually more important: it is error prone. It is easy to create errors that are hard to debug, and even if your notebook works today, it might break easily with future versions of R and the R packages you use. Let me illustrate the problem. 

Suppose you have a date selector on the dashboard and associate a variable with it. You make the questionable decision to call this variable `date`.

Then you want to make sure that `date` has a reasonable value, so in your notebook you set it to today's date:
```
date <<- Sys.Date()
```

You need to put this variable into `.GlobalEnv`, so you need to use the `<<-` operator. This of course fails immediately with the rather cryptic error message:
```
Error: cannot change value of locked binding for 'date'
```
The reason is simple if you know how the `<<-` operator works. It tries to assign to the first occurrence of the specified symbol (`date`), in the search path. If it does not find the symbol, and only then, it assigns to `.GlobalEnv`. If it happens to find the symbol in *any* of the attached packages, then it tries to replace it with the new value.

In our example with `date` it finds `base::date`, so it tries to overwrite that. But the environment of `base` is locked, hence the error message.

Granted, you would not call your variable `date`, but the problem is that there are thousands of exported names from the base packages only, and if you include more packages, there will be even more of them. If you happen to pick any of these names, you'll be in trouble.

Also, the set of exported names is growing over time, so even if your notebook/dashboard works today, it might fail with newer versions of R or the packages you use.

In summary, let's try to get rid of `<<-`. It requires some changes in `RCloud`, but these changes are easy and backward compatible. Nothing will break. 

I'll open an issue for this in the Rcloud repo as well.
